lib/rules/max-lines-per-function.js

```diff
@@ -190,11 +193,21 @@ module.exports = {
 
             if (lineCount > maxLines) {
                 const name = upperCaseFirst(astUtils.getFunctionNameWithKind(funcNode));
+                const linesExceed = lineCount - maxLines;
+
+                const loc = {
+                    start: {
+                        line: maxLines + 1,
+                        column: 0
+                    },
```

Comment:
Sorry if my [previous comment](https://github.com/eslint/eslint/pull/15140#discussion_r740432372) wasn't clear enough.

`line: maxLines + 1` is incorrect, because it reports a line that isn't related to the function:

![image](https://user-images.githubusercontent.com/44349756/140531588-1db5015d-2022-4633-9f1b-7ac27abb66e7.png)

In the example from the screenshot, reported line should be 14, because that's the 6th line of the function:

![image](https://user-images.githubusercontent.com/44349756/140532076-d2be0154-4795-4234-a008-feba7c0d179b.png)

It would be helpful to add some tests cases where the function doesn't start on the first line.

Also, if the rule is configured to ignore blank lines, then a blank line shouldn't be marked as the start of lines that are above the limit, because it doesn't contribute to the line count. We should find the first line that goes above the limit of those that are counted. Same for comments. An example of how this is implemented is [max-lines](https://github.com/eslint/eslint/blob/c415c041912a3abbf106cc5713bdcf4ef42590ac/lib/rules/max-lines.js) rule.

Reply:
It should report only those start and end lines in loc property which start after the maxLine or It should just report the start and end line in loc property from the start and end of the whole function not related to maxLine?

Reply:
It should highlight extra lines in the function. Those are lines that start after the last allowed line in the function.

For example, if the function has 7 lines, and the max is 5, it should highlight the 6th and the 7th line of the function. So, `loc.start.line` should be the 6th line of the function (not the 6th line of the whole file). This is how it should look like:

![image](https://user-images.githubusercontent.com/44349756/140532076-d2be0154-4795-4234-a008-feba7c0d179b.png)

```js
loc: {
    start: {
        line: 14,
        column: 0
    },
    end: {
        line: 15,
        column: 1
    }
}
```

It's a bit more complicated if the rule is configured to skip blank lines and/or comments:

> Also, if the rule is configured to ignore blank lines, then a blank line shouldn't be marked as the start of lines that are above the limit, because it doesn't contribute to the line count. We should find the first line that goes above the limit of those that are counted. Same for comments. An example of how this is implemented is [max-lines](https://github.com/eslint/eslint/blob/c415c041912a3abbf106cc5713bdcf4ef42590ac/lib/rules/max-lines.js) rule.